addons/crm/crm_lead.py

```diff
@@ -547,8 +552,7 @@ def _merge_notify(self, cr, uid, opportunity_id, opportunities, context=None):
         for opportunity in opportunities:
             subject.append(opportunity.name)
             title = "%s : %s" % (opportunity.type == 'opportunity' and _('Merged opportunity') or _('Merged lead'), opportunity.name)
-            fields = list(CRM_LEAD_FIELDS_TO_MERGE)
```

Comment:
Just replace this line by:
fields = self._lead_merge_fields(cr, uid, context=context) 
dont forget context and dont touch the other lines to keep diff minimal.